db.go

```diff
@@ -312,14 +333,33 @@ func (db *DB) retentionCutoff() (b bool, err error) {
 		return false, nil
 	}
 
+	// Size based retention, if MaxBytes is still -1, it has not been set by the user
+	// and only time based retention will be used.
+	var dirsBySize []string
+	if (db.opts.MaxBytes > 0){
+		dirsBySize, err = maxByteCutoffDirs(db.dir, db.opts.MaxBytes, db.metrics)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	if (len(dirsBySize) > 0) {
+        db.metrics.dataLimitDeletions.Inc()
+		level.Warn(db.logger).Log("msg", "data limit was exceeded and records deleted")
+	}
+
+	// Time based retention
 	last := blocks[len(db.blocks)-1]
 
 	mint := last.Meta().MaxTime - int64(db.opts.RetentionDuration)
-	dirs, err := retentionCutoffDirs(db.dir, mint)
+	dirsByTime, err := retentionCutoffDirs(db.dir, mint)
 	if err != nil {
 		return false, err
 	}
 
+	// Remove blocks that fall under either retention criteria
+	dirs := append(dirsBySize, dirsByTime...)
```

Comment:
I think this needs to dedupe blocks in case a block was both too old and needs to be deleted for size too? Otherwise in `reload()` it'll try to delete it twice and log a warning, IIUC.

Reply:
I could dedupe it, but the reload method passes through each block in the directory and then checks if it's in the 'deleteable' list that's passed into reload, and if so then deletes it. So I don't think it should ever try to delete a block twice.